calculator/main.py

```diff
@@ -26,7 +28,19 @@ def perform_operation(calc, operation, params):
     :param params: Tuple containing the list of nums to operate with.
                    ie: (1, 2, 3, 4.5, -2)
     """
-    pass
+    now = time.strftime("%Y-%m-%d %H:%M:%S")
+    if operation == 'plot':
+        return plot_op(*params)
+    try:
+        res = calc['operations'][operation](*params)
+    except KeyError:
+        raise InvalidOperation('"{}" operation not supported.'.format(operation))
+    except ValueError:
+        raise InvalidParams('Given params are invalid.')
```

Comment:
I've seen programs where string variables , like invalid = 'Given params are invalid." are defined, and then those variables are used as messages in exceptions. It might not be as useful in this code, but could find use when the messages raised are bigger.